testing/src/main/java/io/grpc/internal/testing/AbstractTransportTest.java

```diff
@@ -1055,6 +1058,19 @@ public Void answer(InvocationOnMock invocation) throws Exception {
   }
 
   /**
+   * Overriding methods will set non-trivial transport attributes first and then call this method.
+   */
+  protected void clientStreamGetAttributes() throws Exception {
```

Comment:
Let's not put this in the shared code space. It's small and already transport-specific.